homeassistant/components/integration/config_flow.py

```diff
@@ -0,0 +1,81 @@
+"""Config flow for Integration - Riemann sum integral integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any, cast
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_METHOD,
+    CONF_NAME,
+    TIME_DAYS,
+    TIME_HOURS,
+    TIME_MINUTES,
+    TIME_SECONDS,
+)
+from homeassistant.helpers import selector
+from homeassistant.helpers.helper_config_entry_flow import (
+    HelperConfigFlowHandler,
+    HelperFlowStep,
+)
+
+from .const import (
+    CONF_CUSTOM_UNIT_ENABLE,
+    CONF_CUSTOM_UNIT_OF_MEASUREMENT,
+    CONF_ROUND_DIGITS,
+    CONF_SOURCE_SENSOR,
+    CONF_UNIT_PREFIX,
+    CONF_UNIT_TIME,
+    DOMAIN,
+    INTEGRATION_METHODS,
+    METHOD_TRAPEZOIDAL,
+)
+
+UNIT_PREFIXES = ["none", "k", "M", "G", "T"]
+TIME_UNITS = [TIME_SECONDS, TIME_MINUTES, TIME_HOURS, TIME_DAYS]
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_METHOD, default=METHOD_TRAPEZOIDAL): selector.selector(
+            {"select": {"options": INTEGRATION_METHODS}}
+        ),
+        vol.Required(CONF_ROUND_DIGITS, default=2): selector.selector(
+            {"number": {"min": 0, "max": 6, "mode": "box"}}
+        ),
+        vol.Required(CONF_UNIT_PREFIX, default=TIME_HOURS): selector.selector(
+            {"select": {"options": UNIT_PREFIXES}}
+        ),
+        vol.Required(CONF_UNIT_TIME): selector.selector(
+            {"select": {"options": TIME_UNITS}}
+        ),
+        vol.Required(CONF_CUSTOM_UNIT_ENABLE, default=False): selector.selector(
+            {"boolean": {}}
+        ),
+        vol.Required(CONF_CUSTOM_UNIT_OF_MEASUREMENT): selector.selector({"text": {}}),
```

Comment:
Should we really add a deprecated option?

Reply:
Good point @elupus

It was deprecated because it caused a lot of confusion; the custom unit is used for display, but the configured `unit_prefix` and `unit_time` are used for calculation.
In the config flow we could either just not support it, or allow overriding the source sensor's base unit, but use `unit_prefix` and `unit_time` both for display and for calculations.

@dgomes what do you think?

Reply:
I think just not support it..it was intended to be removed by deprecation anyway. To override the base sensor unit, customize/template should be used (or base integration fixed)

Reply:
Agree, config flow should drop the legacy stuff